app/ui/shared/widgets/search.jsx

```diff
@@ -43,6 +43,8 @@ class Search extends Component {
 
     this.state = {
       interaction: INTERACTION_TYPES.IDLE,
+      focused: false,
+      focusByMouse: false,
```

Comment:
When the user is typing, a temporary INTERACTION_TYPE local state is set on the local component state which is assigned as an attribute to the actual input element. In the `setURLBarValue` saga, this attribute is checked before trying to set the url bar value so that the input value isn't changed while the user is typing. Such a behavior makes also sense while the input is being user-selected. So, instead of using `focusByMouse` as yet another local state property, just add another INTERACTION_TYPE, making sure to reset it to IDLE when blurring occurs.